GE05-TEAM6/models/sale_order.py

```diff
@@ -0,0 +1,23 @@
+from odoo import models, api
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.depends('user_id', 'company_id', 'partner_shipping_id')
+    def _compute_warehouse_id(self):
+        sf = ['Alaska', 'California', 'Colorado', 'Hawaii', 'Idaho', 'Montana', 'Nevada', 'Oregon', 'Utah', 'Washington', 
+              'Wyoming', 'Arizona', 'New Mexico', 'Texas', 'Oklahoma', 'Illinois', 'Indiana', 'Iowa', 'Kansas', 
+              'Michigan', 'Minnesota', 'Nebraska', 'South Dakota', 'North Dakota', 'Ohio', 'Wisconsin']
+        bf = ['Alabama', 'Arkansas', 'Florida', 'Georgia', 'Kentucky', 'Louisiana',  'Missouri', 'Mississippi',
+              'South Carolina', 'North Carolina', 'Tennesse', 'Virginia', 'West Virigina', 'Delaware', 'Maryland',
+              'New Jersey', 'Pennsylvania', 'New York', 'Connecticut', 'Maine', 'Massachusetss', 'Vermont', 
+              'New Hampshire', 'Rhode Island']
+
+        for order in self:
```

Comment:
This is fine, but try to filter records, instead of `for order in self:` try something like

```py
for order in self.filtered(lambda o: o.state in ['draft', 'sent']):
```

Reply:
This way you can avoid the 'if' statement in line 17.

Reply:
The reason why we want to do this is because Python is terribly slow with for loops, so we should try to reduce as much as possible the number of records we're iterating through.